crates/solana-gateway-task-processor/src/component.rs

```diff
@@ -659,3 +660,581 @@ async fn send_transaction(
+    use futures::{SinkExt as _, StreamExt as _};
+    use interchain_token_transfer_gmp::{
+        DeployInterchainToken, GMPPayload, InterchainTransfer, ReceiveFromHub,
+    };
+    use pretty_assertions::assert_eq;
+    use relayer_amplifier_api_integration::{
+        AmplifierCommand, AmplifierCommandClient, AmplifierTaskReceiver,
+    };
+    use relayer_amplifier_state::State;
+    use solana_client::nonblocking::rpc_client::RpcClient;
+    use solana_client::rpc_client::GetConfirmedSignaturesForAddress2Config;
+    use solana_client::rpc_config::RpcTransactionConfig;
+    use solana_rpc::rpc::JsonRpcConfig;
+    use solana_rpc::rpc_pubsub_service::PubSubConfig;
+    use solana_sdk::account::AccountSharedData;
+    use solana_sdk::commitment_config::{CommitmentConfig, CommitmentLevel};
+    use solana_sdk::pubkey::Pubkey;
+    use solana_sdk::signature::{Keypair, Signature};
+    use solana_sdk::signer::Signer as _;
+    use solana_sdk::{bpf_loader_upgradeable, keccak, system_program};
+    use solana_test_validator::UpgradeableProgramInfo;
+    use solana_transaction_status::option_serializer::OptionSerializer;
+    use solana_transaction_status::{
+        EncodedConfirmedTransactionWithStatusMeta, UiTransactionEncoding,
+    };
+    use tokio::task::JoinHandle;
+
+    use super::SolanaTxPusher;
+    use crate::config;
+
+    const ITS_HUB_CHAIN_NAME: &str = "axelar";
+    const ITS_HUB_SOURCE_ADDRESS: &str =
+        "axelar157hl7gpuknjmhtac2qnphuazv2yerfagva7lsu9vuj2pgn32z22qa26dk4";
+    const TEST_TOKEN_NAME: &str = "MyToken";
+    const TEST_TOKEN_SYMBOL: &str = "MTK";
+
+    #[test_log::test(tokio::test)]
+    async fn process_successfull_token_deployment() {
+        let mut fixture = setup().await;
+        let (gas_config, _gas_init_sig, _counter_pda, _init_memo_sig, _init_its_sig) =
+            setup_aux_contracts(&mut fixture).await;
+        let (pusher_task, mut task_sender, mut rx_amplifier, _) =
+            setup_tx_pusher(&fixture, &gas_config);
+        let token_id = axelar_solana_its::interchain_token_id(&fixture.payer.pubkey(), b"whatever");
+        let deploy_interchain_token_message =
+            GMPPayload::DeployInterchainToken(DeployInterchainToken {
+                selector: 1_u32.try_into().unwrap(),
+                token_id: token_id.into(),
+                name: TEST_TOKEN_NAME.to_owned(),
+                symbol: TEST_TOKEN_SYMBOL.to_owned(),
+                decimals: 9,
+                minter: fixture.payer.pubkey().to_bytes().into(),
+            });
+
+        send_its_message(
+            deploy_interchain_token_message,
+            ITS_HUB_CHAIN_NAME.to_owned(),
+            ITS_HUB_SOURCE_ADDRESS.to_owned(),
+            &mut fixture,
+            &mut task_sender,
+        )
+        .await;
+
+        task_sender.close_channel();
+        let _result = pusher_task.await;
+
+        assert_eq!(rx_amplifier.next().await, None);
```

Comment:
Can you also add an assert for validating that the state on Solana has changed? (eg the expected PDAs have been initialised). This will give us guarantee that we actually send a tx to the chain